DOC-249 Shopify Plugin (#347)
amplitude/amplitude-docs: content/collections/data/en/amplitude-shopify-plugin.md

---
id: d7658703-c1c3-4cc4-a5e3-3b5674e41405
blueprint: data
title: 'Amplitude Shopify Plugin'
landing: false
exclude_from_sitemap: false
updated_by: 0c3a318b-936a-4cbd-8fdf-771a90c297f0
updated_at: 1729791421
---
The [Amplitude Shopify Plugin](https://apps.shopify.com/amplitude) enables you to instrument your Shopify store with an Amplitude features including [Autocapture](/docs/data/autocapture) and [Session Replay](/docs/session-replay).

> 

Reviewer: ### About Shopify
[Shopify](https://www.shopify.com/) is an all-in-one commerce platform that allows businesses of any size to create, customize, and manage online stores with ease. It offers tools for product listings, payments, shipping, and customer engagement, streamlining the selling process online, across social media, and in person. With built-in marketing features, analytics, and integrations with third-party apps, Shopify helps business owners track performance and drive sales.

Reviewer: ### Considerations

- **Performance**: The packages we utilize on the frontend is roughly 167kb. To give a rough idea of the performance impact I generated 6 different Lighthouse Performance reports. 3 with the Amplitude packages, and 3 without. Without the Amplitude packages, the average performance score was 98. With the packages, the average Performance score was 96.
- **Session replay may not display as expected**. Shopify’s UI heavily relies on animations, specifically the fade-in animation. The animation responsible for changing the opacity of the UI from 0 to 1 is not consistently picked up by session replay, which is why elements appear to be missing.

> The Shopify Plugin plugin installs a version of the [Browser SDK](/docs/sdks/analytics/browser/browser-sdk-2), and adds the script before the `</head>` tag on each of your site's pages.

Reviewer 2: ```suggestion
The Shopify plugin installs a version of the [Browser SDK](/docs/sdks/analytics/browser/browser-sdk-2), and adds the script before the `</head>` tag on each of your site's pages.
```

> The Shopify Plugin plugin installs a version of the [Browser SDK](/docs/sdks/analytics/browser/browser-sdk-2), and adds the script before the `</head>` tag on each of your site's pages.

Reviewer: Amplitude also sends all [Default Events](https://amplitude.com/docs/sdks/analytics/browser/migrate-from-marketing-analytics-to-browser-sdk-2-0#default-event-tracking).

By default, the Shopify plugin captures the following event properties:

- `[Amplitude] Quantity`
- `[Amplitude] SKU`
- `[Amplitude] Price`
- `[Amplitude] Currency Code`
- `[Amplitude] Type`
- `[Amplitude] Variant Title`
- `[Amplitude] Title`
- `[Amplitude] Vendor`
- `[Amplitude] Products`
  - `[Amplitude] Quantity`
  - `[Amplitude] SKU`
  - `[Amplitude] Price`
  - `[Amplitude] Currency Code`
  - `[Amplitude] Type`
  - `[Amplitude] Variant Title`
  - `[Amplitude] Title`
  - `[Amplitude] Vendor`

## Install the plugin

To add the plugin to your Shopify store:

> 1. Log in to your Shopify account, and find the plugin in the [Shopify App Store](https://apps.shopify.com/amplitude). Click *Install*.

Reviewer 2: ```suggestion
1. Log in to your Shopify account and find the plugin in the [Shopify App Store](https://apps.shopify.com/amplitude). Click *Install*.
```

> 2. Confirm the required plugin permissions, and click *Install*.

Reviewer 2: ```suggestion
2. Confirm the required plugin permissions and click *Install*.
```

> 3. Locate your project's API key, and add it to the Amplitude Settings page in Shopify. Click *Connect*.

Reviewer 2: ```suggestion
3. Locate your project's API key and add it to the Amplitude Settings page in Shopify. Click *Connect*.
```

    {{partial:admonition type="note" heading="Ad blocking software"}}

>     Disable any enabled ad blocking software in your browser to ensure the plugin setup step can communicate with Amplitude.

Reviewer 2: ```suggestion
    If any ad blocking software is running in your browser, disable it to ensure the plugin setup step can communicate with Amplitude.
```

    {{/partial:admonition}}

> 4. A "Connection successful" message indicates the plugin is enabled and connected with Amplitude.

Reviewer 2: ```suggestion
A "Connection successful" message indicates the plugin is enabled and connected with Amplitude.
```

Reviewer 2: This isn't a step—the user doesn't do anything here—so it doesn't get a number.

---
id: d7658703-c1c3-4cc4-a5e3-3b5674e41405
blueprint: data
title: 'Amplitude Shopify Plugin'
landing: false
exclude_from_sitemap: false
updated_by: 0c3a318b-936a-4cbd-8fdf-771a90c297f0
updated_at: 1729791421
---

[Shopify](https://www.shopify.com/) is an all-in-one commerce platform that allows businesses of any size to create, customize, and manage online stores with ease. It offers tools for product listings, payments, shipping, and customer engagement, streamlining the selling process online, across social media, and in person. With built-in marketing features, analytics, and integrations with third-party apps, Shopify helps business owners track performance and drive sales.

The [Amplitude Shopify Plugin](https://apps.shopify.com/amplitude) enables you to instrument your Shopify store with an Amplitude features including [Autocapture](/docs/data/autocapture) and [Session Replay](/docs/session-replay).

## Considerations

> * Performance: The packages Amplitude provides are roughly 167kb total size. In testing, Shopify stores with Amplitude installed registered an average lighthouse score of 96, compared to an average score of 98 without Amplitude.

Reviewer 3: ```suggestion
* Performance: The packages Amplitude provides are roughly 167kb total size, but will vary depending on which products you utilize. In testing, Shopify stores with Amplitude installed registered an average lighthouse score of 96, compared to an average score of 98 without Amplitude.
```

> * Session replay may not display as expected. Shopify’s UI relies on animations, specifically the fade-in animation. Session Replay doesn't capture the animation consistently, which may cause elements to appear missing.

Reviewer 3: I think this is okay to take out. This ended up being an issue with my test store, and we haven't received reports from customers that they are running into this

## Plugin details

The Shopify plugin installs a version of the [Browser SDK](/docs/sdks/analytics/browser/browser-sdk-2), and adds the script before the `</head>` tag on each of your site's pages.

By default, the Shopify plugin captures the following event properties:

- `[Amplitude] Quantity`
- `[Amplitude] SKU`
- `[Amplitude] Price`
- `[Amplitude] Currency Code`
- `[Amplitude] Type`
- `[Amplitude] Variant Title`
- `[Amplitude] Title`
- `[Amplitude] Vendor`
- `[Amplitude] Products`
  - `[Amplitude] Quantity`
  - `[Amplitude] SKU`
  - `[Amplitude] Price`
  - `[Amplitude] Currency Code`
  - `[Amplitude] Type`
  - `[Amplitude] Variant Title`
  - `[Amplitude] Title`
  - `[Amplitude] Vendor`

## Install the plugin

To add the plugin to your Shopify store:

1. Log in to your Shopify account and find the plugin in the [Shopify App Store](https://apps.shopify.com/amplitude). Click *Install*.
2. Confirm the required plugin permissions and click *Install*.

> 3. Locate your project's API key and add it to the Amplitude Settings page in Shopify. Click *Connect*.

Reviewer 3: We also just added a step where a user needs to indicate where they want to store their data. 
![image](https://github.com/user-attachments/assets/55dd6837-bcbc-4ae5-a767-8ded81dbca12)
